### PR TITLE
Feat/#309 read config from arbitrary directories

### PR DESCRIPTION
**Pull request by jiachen1120**
Related issue: https://github.com/networknt/light-4j/issues/309
Related RFC: https://github.com/networknt/light-rfcs/blob/master/light-4j/0012-flexible-config-directory.md

Three new overloaded methods provided, in this PR, the path can only be the path relative to light-4j-config-dir. 
public abstract Map<String, Object> getJsonMapConfig(String configName, String path);
public abstract Map<String, Object> getJsonMapConfigNoCache(String configName, String path);
public abstract Object getJsonObjectConfig(String configName, Class clazz, String path);

Extended ConfigDefaultTest to included the test for these methods. The ConfigDefaultTest could not work properly before if build the whole light-4j project entirely. Since the config is a singleton and shared by all classes, the instance of config would already exist before running ConfigDefaultTest. If we set the light-4j-config-dir directly, it cannot be injected into config instance. So the config in ConfigDefaultTest would still be the classpath one. So now I use reflection to inject field in the test.

Look forward to your review!

### CODE REVIEW

File: config/src/main/java/com/networknt/config/ConfigInjection.java [hunk 1]
```diff
@@ -160,4 +169,29 @@ public void setKey(String key) {
...
     }
+
+    // Method used to cast string into int, double or boolean
+    private static Object typeCast(String str) {
+        if (str == null || str.equals("")) {
+            return null;
+        }
+        if (isBoolean(str)) {
+            return str.equals("true") || str.equals("True") || str.equals("TRUE") ? true : false;
+        }
```
Comment (NicholasAzar) [inline config/src/main/java/com/networknt/config/ConfigInjection.java]: Can be simplified to just `return str.equals("true") || str.equals("True") || str.equals("TRUE");` right?

Comment (miklish) [reply config/src/main/java/com/networknt/config/ConfigInjection.java]: or
return str.toLowerCase().equals("true");

Comment (jiachen1120) [reply config/src/main/java/com/networknt/config/ConfigInjection.java]: Yes, I will simplify it. @miklish The reason why I didn't use toLowerCase() is to uniform with yaml rules, ie "tRue", "trUe", etc. should not be recognized as boolean. Do you think it makes sense?

Comment (miklish) [reply config/src/main/java/com/networknt/config/ConfigInjection.java]: It simply depends on how strict you want your parsing to be in this instance.

Comment (NicholasAzar) [reply config/src/main/java/com/networknt/config/ConfigInjection.java]: @jiachen1120 Agreed, we shouldn't use toLowerCase.
Actually, if you want to be even closer to the [yaml boolean spec](https://yaml.org/type/bool.html), we would use these:
```
 y|Y|yes|Yes|YES|n|N|no|No|NO
|true|True|TRUE|false|False|FALSE
|on|On|ON|off|Off|OFF
```

Comment (jiachen1120) [reply config/src/main/java/com/networknt/config/ConfigInjection.java]: Sure, I will improve it to be more strict.

Comment (jiachen1120) [reply config/src/main/java/com/networknt/config/ConfigInjection.java]: And I will modify the light-doc as well.

**Pull request merged**